### PR TITLE
As Hubs can be registered, the same should be available for Persistent connections

### PR DESCRIPTION
**Pull request by vladmir-boy**

### CODE REVIEW

File: src/Autofac.Integration.SignalR/RegistrationExtensions.cs [hunk 1]
```diff
@@ -48,5 +50,18 @@ public static class RegistrationExtensions
                 .Where(t => typeof(IHub).IsAssignableFrom(t))
                 .ExternallyOwned();
         }
+
+        /// <summary>
+        /// Register types that implement <see cref="PersistentConnection"/> in the provided assemblies.
+        /// </summary>
+        /// <param name="builder">The container builder.</param>
+        /// <param name="controllerAssemblies">Assemblies to scan for controllers.</param>
```
Comment (alexmg) [inline src/Autofac.Integration.SignalR/RegistrationExtensions.cs]: "controllers" should be "persistent connections" in comments

File: src/Autofac.Integration.SignalR/RegistrationExtensions.cs [hunk 2]
```diff
@@ -48,5 +50,18 @@ public static class RegistrationExtensions
...
                 .ExternallyOwned();
         }
+
+        /// <summary>
+        /// Register types that implement <see cref="PersistentConnection"/> in the provided assemblies.
+        /// </summary>
+        /// <param name="builder">The container builder.</param>
+        /// <param name="controllerAssemblies">Assemblies to scan for controllers.</param>
+        /// <returns>Registration builder allowing the controller components to be customised.</returns>
+        public static IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle> RegisterConnections(this ContainerBuilder builder, params Assembly[] controllerAssemblies)
```
Comment (alexmg) [inline src/Autofac.Integration.SignalR/RegistrationExtensions.cs]: Perhaps rename method to "RegisterPersistentConnections" to be more specific. I think the parameter name should just be "assemblies". The parameter name for the RegisterHubs method shouldn't have been called "controllerAssemblies" either (not your mistake though).

**Comment by alexmg** [comment]
Naming issues aside, it looks like a useful addition. Is this something you have actually been using yourself?

**Comment by vladmir-boy** [comment]
Yes, since we don't use hubs (we don't need to broadcast, we have only one-to-one connections) we want to use lower-level connections and have the possibility to inject into

**Comment by tillig** [comment]
Pulled into the (newly created) `develop` branch. Thanks!

**Pull request closed without merging**